backend/app/models/user.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'bcrypt'
+require 'securerandom'
+
+# Create user model
+class User < ApplicationRecord
+  attr_reader :password
+
+  validates :username, presence: true, uniqueness: true
+  validates :email, presence: true, uniqueness: true
+  validates :password_digest, :session_token, presence: true
+  validates :password, length: { minimum: 6 }, allow_nil: true
+
+  after_initialize :ensure_session_token
+
+  has_many :favorites,
+           primary_key: :id,
+           foreign_key: :user_id
+
+  has_many :reservations,
```

Comment:
I don’t think any of the has_many associations are relevant to this PR. remove please

Reply:
Okay, have removed associations